src/EditorFeatures/Core.Wpf/Adornments/AbstractAdornmentManagerProvider.cs

```diff
@@ -35,20 +35,6 @@ internal abstract class AbstractAdornmentManagerProvider<TTag> :
         protected abstract string FeatureAttributeName { get; }
         protected abstract string AdornmentLayerName { get; }
 
-        public void TextViewCreated(IWpfTextView textView)
-        {
-            if (textView == null)
-            {
-                throw new ArgumentNullException(nameof(textView));
-            }
-
-            if (!textView.TextBuffer.GetFeatureOnOffOption(EditorComponentOnOffOptions.Adornment))
-            {
-                return;
-            }
```

Comment:
this is code you likely want to run in any subclass.  so instead of this pattern, have a non virtual TextViewCreated that does this logic.  Then have it call into a new protected abstract method that all subclasses have to implement that then doesn't need to duplciate this logic.

Reply:
fixed